pandas/tests/base/test_conversion.py

```diff
@@ -444,3 +444,11 @@ def test_to_numpy_dataframe_single_block(data, expected):
     df = pd.DataFrame(data)
     result = df.to_numpy(dtype=float, na_value=np.nan)
     tm.assert_numpy_array_equal(result, expected)
+
+
+def test_to_numpy_dataframe_single_block_no_mutate():
+    # https://github.com/pandas-dev/pandas/issues/33820
+    result = pd.DataFrame(np.array([1.0, 2.0, np.nan]))
+    expected = pd.DataFrame(np.array([1.0, 2.0, np.nan]))
+    result.to_numpy(dtype=float, na_value=0.0)
```

Comment:
```suggestion
    result.to_numpy(na_value=0.0)
```

I think this might trigger editing the original

Reply:
Ah, you're right. I guess we do need to check for this explicitly.